docs/design.md

```diff
@@ -3,22 +3,65 @@ CStor Pool Auto controller will consist of following deployments:
+cstorpoolauto controller watches Kubernetes resource(s) and invokes appropriate http endpoints exposed by cspauto service. In other words, this is the http client for cstorpoolauto service.
 
 ### Conformance Deployment (optional)
 Conformance deployment has the conformance logic to verify if cspauto service is working as expected. This is completely optional and can be installed or un-installed without any disruptions to the core services. Conformance deployment will consist of following components:
-- cspauto-conformance (a conformance http service)
-- cspauto-conformance-controller (a http client & kubernetes controller)
+- cstorpoolauto-conformance (a conformance http service)
+- cstorpoolauto-conformance-controller (a http client & kubernetes controller)
 
-#### cspauto-conformance
-cspauto conformance implements conformance logic to verify if cspauto service is functioning properly. This is deployed as a http service. This exposes one or more http endpoints that get invoked by cspauto conformance controller.
+#### 1/ cstorpoolauto-conformance
+cstorpoolauto conformance implements conformance logic to verify if cstorpoolauto service is functioning properly. This is deployed as a http service. This exposes one or more http endpoints that get invoked by cstorpoolauto conformance controller.
 
-#### cspauto-conformance-controller
-cspauto conformance controller watches Kubernetes resource(s) and invokes appropriate http endpoints exposed by cspauto conformance service. In other words, this is the http client for cspauto conformance service.
\ No newline at end of file
+#### 2/ cstorpoolauto-conformance-controller
+cstorpoolauto conformance controller watches Kubernetes resource(s) and invokes appropriate http endpoints exposed by cstorpoolauto conformance service. In other words, this is the http client for cstorpoolauto conformance service.
+
+### Custom Resource Definitions
+- CSPCAuto handles cstor pool related operations in an automated manner.
+```yaml
+kind: CSPCAuto
+metadata:
+    name:
+    annotations:
+        # These annotations that determine the storage provider.
+        # They are required for create, delete, attach, etc
+        # storage operations.
+        dao.mayadata.io/storage-class-name: 
+        dao.mayadata.io/csi-driver-name:
+spec:
+    # stripe, mirror, raidz
+    #
+    # this plays a role to select the number of storage
+    # instances needed on selected nodes
+    poolType:
+
+    # storage capacity to be used during storage creation
+    createCapacity:
+
+    # Node(s) will be selected based on the provided
+    # selector terms
+    #
+    # A selector term will AND its match fields to select
+    # a node from the cluster. Similarly list of selector
+    # terms will be ORed to arrive at the final set of
+    # eligible nodes from the cluster.
+    #
+    # Controller will validate if these selected nodes 
+    # are capable enough to provision pool based on the 
+    # pool type.
+    nodeSelector:
```

Comment:
Do we reconcile for any changes here?
how do we allow to add new node that doesn't match given nodeSelector?
Similar for disk selector as well if one such thing exists

Reply:
I guess we can avoid reconcile here.
Let us rename this to `CSPCMake`. This will be one such CR among a list of others that make up `CstorPoolAuto` operations.